### PR TITLE
[SPARK-24038][SS] Refactor continuous writing to its own class

### PR DESCRIPTION
**Pull request by jose-torres**
## What changes were proposed in this pull request?

Refactor continuous writing to its own class.

See WIP https://github.com/jose-torres/spark/pull/13 for the overall direction this is going, but I think this PR is very isolated and necessary anyway.

## How was this patch tested?

existing unit tests - refactoring only

### CODE REVIEW

**Comment by jose-torres** [comment]
@tdas

**Review by xuanyuanking — COMMENTED** [review]
Cool refactor, more clear logic of CP plans, just a nit.

**Comment by tdas** [comment]
LGTM!

File: sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/continuous/WriteToContinuousDataSourceExec.scala [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+  override def output: Seq[Attribute] = Nil
+
+  override protected def doExecute(): RDD[InternalRow] = {
+    val writerFactory = writer match {
+      case w: SupportsWriteInternalRow => w.createInternalRowWriterFactory()
+      case _ => new InternalRowDataWriterFactory(writer.createWriterFactory(), query.schema)
+    }
+
+    val rdd = query.execute()
+    val messages = new Array[WriterCommitMessage](rdd.partitions.length)
```
Comment (tdas) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/continuous/WriteToContinuousDataSourceExec.scala]: Is this really needed. The only use of it is in the logInfo before, that too, only in the length, which is effectively `rdd.partitions.length`.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/continuous/WriteToContinuousDataSourceExec.scala [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+object WriteToContinuousDataSourceExec extends Logging {
+  def run(
+      writeTask: DataWriterFactory[InternalRow],
+      context: TaskContext,
+      iter: Iterator[InternalRow]): Unit = {
+    val epochCoordinator = EpochCoordinatorRef.get(
+      context.getLocalProperty(ContinuousExecution.EPOCH_COORDINATOR_ID_KEY),
+      SparkEnv.get)
+    val currentMsg: WriterCommitMessage = null
```
[the remark this answers is not in the document]
Comment (tdas) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/continuous/WriteToContinuousDataSourceExec.scala]: Good point. I see its no long used anywhere. That raises two questions. 
- This should be removed. 
- Does this continuous code path not have to send back a WriterMessage? How is that working?

[the remark this answers is not in the document]
Comment (tdas) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/continuous/WriteToContinuousDataSourceExec.scala]: Nvm. I see the msg being sent back using the epochCoordinator. Then lets just remove the `currentMsg`

**Comment by jose-torres** [comment]
addressed comments

**Pull request closed without merging**